packages/core/contracts/financial-templates/long-short-pair/LongShortPair.sol

```diff
@@ -223,23 +223,20 @@ contract LongShortPair is Testable, Lockable {
      */
     function settle(uint256 longTokensToRedeem, uint256 shortTokensToRedeem)
         public
-        postExpiration()
         nonReentrant()
         returns (uint256 collateralReturned)
     {
-        // If the contract state is open and postExpiration passed then `expire()` has not yet been called.
-        require(contractState != ContractState.Open, "Unexpired contract");
-
-        // Get the current settlement price and store it. If it is not resolved, will revert.
-        if (contractState != ContractState.ExpiredPriceReceived) {
-            expiryPrice = _getOraclePriceExpiration(expirationTimestamp);
-            // Cap the return value at 1.
-            expiryPercentLong = Math.min(
-                financialProductLibrary.percentageLongCollateralAtExpiry(expiryPrice),
-                FixedPoint.fromUnscaledUint(1).rawValue
-            );
-            contractState = ContractState.ExpiredPriceReceived;
-        }
+        // Either time must have passed the expirationTimestamp (normal expiration) or it is before expirationTimestamp
+        // and early expiration is enabled enabled.
+        require(
+            getCurrentTime() > expirationTimestamp || (getCurrentTime() < expirationTimestamp && enableEarlyExpiration),
```

Comment:
nit: it's more readable IMO to have `enableEarlyExpiration` appear first in the && block of the RHS of the || block. It also might exit faster on average since that bool is easier to check than the "<" comparison.

Reply:
sure.